osu.Game/Skinning/Editor/SkinBlueprint.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Edit;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Screens.Play.HUD;
+using osuTK;
+
+namespace osu.Game.Skinning.Editor
+{
+    public class SkinBlueprint : SelectionBlueprint<ISkinnableComponent>
```

Comment:
Which shared code? One is handling a `DrawableHitObject` and the other is handling a `Drawable`. How would you see the abstraction working?

Reply:
Oh never mind me, I've been confused by `ISkinnableComponent` for whether it's an item or a drawable item, and that it's both at once, unlike hit objects and their drawable representations where each is separated.

Can't make an abstraction to fit both of these.

Reply:
As an aside, editor blueprints used to be based around the `DrawableHitObject`. With the introduction of this abstracted editor, it may make more sense to go back to that, which is something I have on my todo list as a research task.

Reply:
Sounds like doing that might get hairy with rulesets that use pooling, but I can't say for sure how problematic it'll be.